.gitlab/functional_test/system_probe.yml

```diff
@@ -11,17 +11,26 @@
     - .kitchen_datadog_agent_flavor
   rules:
     !reference [.on_system_probe_changes_or_manual]
+  needs:
+    job: get_file_diff
+    artifacts: true
   stage: functional_test
   variables:
     AGENT_MAJOR_VERSION: 7
     DD_PIPELINE_ID: $CI_PIPELINE_ID-fnct
     CHEF_VERSION: 14.15.6
   script:
+    - set +e
+    - echo "$DIFF_FILES" | grep "pkg/collector/corechecks/ebpf/" || echo "$DIFF_FILES" | grep "pkg/ebf/" || echo "$DIFF_FILES" | grep "pkg/network/" || echo "$DIFF_FILES" | grep "pkg/util/kernel" || echo "$DIFF_FILES" | grep "test/kitchen/site-cookbooks/dd-system-probe-check/" || echo "$DIFF_FILES" | grep "test/kitchen/test/integration/system-probe-test/" || echo "$DIFF_FILES" | grep "test/kitchen/test/integration/win-sysprobe-test/" || echo "$DIFF_FILES" | grep ".gitlab/functional_test/system_probe.yml" || echo "$DIFF_FILES" | grep "test/new-e2e/system-probe/" || echo "$DIFF_FILES" | grep "test/new-e2e/scenarios/system-probe/"; export exit_code=$?
```

Comment:
I think this would be more readable if you put the path match list in a variable and this whole bit for matching into a separate script that you can reference from multiple job definitions. You can loop over the variable in the script instead of listing out each individual grep.

Reply:
I agree it's a way more cleaner way. I'm going to try that 👍

Reply:
Should be better now :)